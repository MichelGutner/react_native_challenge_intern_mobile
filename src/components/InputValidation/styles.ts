import styled from 'styled-components/native';

export const Container = styled.View`
  flex-direction: row;
  margin-top: 16px;
  width: 322px;
  height: 48px;
  background-color: #0050f0;
  border: 1px solid #ffffff;
  border-radius: 4px;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  display: flex;
  align-items: center;
  text-align: center;
  color: white;
`;

export const Input = styled.TextInput`
  width: 290px;
  height: 48px;
  border-radius: 4px;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  display: flex;
  align-items: center;
  text-align: center;
  color: white;
`;
