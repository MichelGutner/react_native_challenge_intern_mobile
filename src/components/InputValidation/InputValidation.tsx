import React from 'react';
import Icon from 'react-native-vector-icons/AntDesign';
import SignInputs from '../../components/Inputs/SignInputs';
import { Container } from './styles';

type Props = {
  value: string;
  onChangeText: (value: string) => void;
  secureText: boolean;
  isInputValid: boolean;
  placeholder: string;
  placeholderTextColor: string;
};

export const InputValidation = ({
  value,
  placeholder,
  placeholderTextColor,
  onChangeText,
  secureText,
  isInputValid,
}: Props) => {
  return (
    <Container>
      <SignInputs
        placeholderTextColor={placeholderTextColor}
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        secureTextEntry={secureText}
      />
      {isInputValid ? (
        <Icon
          style={{ marginTop: 20 }}
          name={'check'}
          size={20}
          color="green"
        />
      ) : (
        <Icon style={{ marginTop: 20 }} name={'close'} size={20} color="red" />
      )}
    </Container>
  );
};
