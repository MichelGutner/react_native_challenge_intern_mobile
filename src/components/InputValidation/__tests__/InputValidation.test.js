import React from 'react';
import renderer from 'react-test-renderer';
import { InputValidation } from '../InputValidation';

test('renders correctly', () => {
  const tree = renderer.create(<InputValidation />).toJSON();
  expect(tree).toMatchSnapshot();
});
