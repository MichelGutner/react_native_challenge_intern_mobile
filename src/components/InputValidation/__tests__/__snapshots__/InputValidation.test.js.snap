// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0050f0",
        "borderColor": "#ffffff",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "white",
        "display": "flex",
        "flexDirection": "row",
        "fontFamily": "Roboto",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "height": 48,
        "marginTop": 16,
        "textAlign": "center",
        "width": 322,
      },
    ]
  }
>
  <TextInput
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "color": "white",
          "display": "flex",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "height": 48,
          "textAlign": "center",
          "width": 290,
        },
      ]
    }
  />
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î˜³
  </Text>
</View>
`;
